components/button/src/split-button/split-button.js

```diff
@@ -18,8 +18,25 @@ class SplitButton extends Component {
     state = {
         open: false,
     }
+
     anchorRef = React.createRef()
 
+    componentDidMount() {
+        document.addEventListener('keydown', this.handleKeyDown)
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('keydown', this.handleKeyDown)
+    }
+
+    handleKeyDown = (event) => {
+        if (event.key === 'Escape' && this.state.open) {
```

Comment:
I'm pretty sure we should stop event propagation here. See how it's done in eg. `Select` component. 

```
handleKeyDown = (event) => {
        if (!e.defaultPrevented && event.key === 'Escape' && this.state.open) {
            event.stopPropagation()
            this.setState({ open: false })

            this.anchorRef.current && this.anchorRef.current.focus()
        }
    }
```

Maybe it would make sense to add test case for this in your modal-PR. Add dropdown buttons and split-buttons - open them - then click escape, and check that the modal is not closed.